frontend/packages/console-shared/src/components/markdown-highlight-extension/MarkdownHighlightExtension.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+import { extension } from 'showdown';
+import { Spotlight } from '../spotlight';
+
+extension('quickstart', () => {
+  return [
+    {
+      type: 'lang',
+      regex: /\[(\w+)]{{(highlight)\s([\w-]+)}}/g,
+      replace: (text: string, linkLabel: string, linkType: string, linkId: string): string => {
+        return `<button class="pf-c-button pf-m-inline pf-m-link" data-highlight="${linkId}">${linkLabel}</button>`;
+      },
+    },
+  ];
+});
+type MarkdownHighlightExtensionProps = {
+  docContext: HTMLDocument;
+};
+const MarkdownHighlightExtension: React.FC<MarkdownHighlightExtensionProps> = ({ docContext }) => {
+  const [selector, setSelector] = React.useState<string>(null);
+  React.useEffect(() => {
+    const elements = docContext.querySelectorAll('[data-highlight]');
+    function startHighlight(e) {
+      const highlightId = e.target.getAttribute('data-highlight');
+      if (!highlightId) {
+        return;
+      }
+      setSelector(null);
+      setTimeout(() => {
+        setSelector(`[data-tour-id="${highlightId}"]`);
```

Comment:
We need to settle on the correct data marker for what we allow to be targeted by the highlight.
It shouldn't be arbitrary.

We also need to sanitize the `highlightId` otherwise someone could create a selector that closes the first rule and opens another.

Reply:
Based on the regex of the extension, it will already ensure that the highlight ID is `[\w-]+`. Therefore at this time I don't think any further sanitation is necessary.
When we start to support more actions and if we make the regex more generic, we may need to revisit the sanitizing here.